cpp/ql/src/Likely Bugs/Likely Typos/FutileParams.ql

```diff
@@ -12,11 +12,16 @@
 
 import cpp
 
-from Function f, FunctionCall fc
-where fc.getTarget() = f
-  and f.getNumberOfParameters() = 0
-  and not f.isVarargs()
-  and fc.getNumberOfArguments() != 0
-  and not f instanceof BuiltInFunction
-  and exists(FunctionDeclarationEntry fde | fde = f.getADeclarationEntry() | not fde.isImplicit())
-select fc, "This call has arguments, but $@ is not declared with any parameters.", f, f.toString()
+from FunctionCall fc, Function f
+  where f = fc.getTarget() and not f.isVarargs()
+  /* There must be a zero-parameter declaration */
+  and exists ( FunctionDeclarationEntry fde | fde = f.getADeclarationEntry() | fde.getNumberOfParameters() = 0)
+  /* There must be a mismatch between number of call arguments and number of parameters in some
+   * declaration of Function f
+   */
+  and exists ( FunctionDeclarationEntry fde | fde = f.getADeclarationEntry() | fde.getNumberOfParameters() != fc.getNumberOfArguments())
+  /* There must be no actual declaration of Function f whose number of parameters matches number of call arguments */
+  and not exists ( FunctionDeclarationEntry fde | fde = f.getADeclarationEntry() | not fde.isImplicit() and fde.getNumberOfParameters() = fc.getNumberOfArguments())
+select fc, "This call has arguments, but $@ is not declared with any parameters.", f, f.toString()  
+
+  
```

Comment:
Please autoformat (Ctrl+Shift+F). I think you'll get better results from the auto-formatter if you use `//` for in-line comments instead of `/* */`.